[quant][docs] Add docstrings for operators defined in torch.ops.quantized_decomposed namespace

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #89547

Summary:
no functionality changes

Test Plan:
NA

Reviewers:

Subscribers:

Tasks:

Tags:

cc @jianyuh @raghuramank100 @jamesr66a @vkuzo @jgong5 @Xia-Weiwen @leslie-fang-intel

jerryzh168 committed:
[quant][docs] Add docstrings for operators defined in torch.ops.quantized_decomposed namespace

Summary:
no functionality changes

Test Plan:
NA

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented:
@pytorchbot merge

vkuzo commented on torch/ao/quantization/fx/_decomposed.py:
```diff
@@ -32,6 +32,20 @@ def _quant_min_max_bounds_check(quant_min, quant_max, dtype):
 
 @impl(quantized_decomposed_lib, "quantize_per_tensor", "CompositeExplicitAutograd")
 def quantize_per_tensor(input, scale, zero_point, quant_min, quant_max, dtype):
+    """ Affine quantization for the Tensor using the same quantization parameters to map
```
can we add types, both annotated in the code and also in the docstring?  For example, it's not clear if `scale` is a scalar or a tensor.

vkuzo commented on torch/ao/quantization/fx/_decomposed.py:
```diff
@@ -32,6 +32,20 @@ def _quant_min_max_bounds_check(quant_min, quant_max, dtype):
+       dtype: requested dtype for output Tensor
+
+    Returns:
+       per tensor quantized Tensor with requested dtype
```
is this true?  I thought this returns a tensor of integers, which need to be combined with scale and zero_point to interpret the values.  "per tensor quantized tensor" sounds like "tensor + scale + zero_point", which doesn't match the function

jerryzh168 replied:
oh, makes sense, I'll explain this in a clearer way

jerryzh168 committed:
Update on "[quant][docs] Add docstrings for operators defined in torch.ops.quantized_decomposed namespace"


Summary:
no functionality changes

Test Plan:
NA

Reviewers:

Subscribers:

Tasks:

Tags:

cc jianyuh raghuramank100 jamesr66a vkuzo jgong5 Xia-Weiwen leslie-fang-intel

[ghstack-poisoned]

jerryzh168 commented:
@pytorchbot merge -f "broken tests unrelated to the PR"
